### PR TITLE
[Merged by Bors] - perf(NormedSpace/OperatorNorm): fix `simp` call and clean up porting notes

### PR DESCRIPTION
**Pull request by mattrobball**
We clean up some porting notes and speed up this file.

---

- [ ] depends on: #10010 
- [ ] depends on: #10012

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

**Comment by mattrobball** [comment]
!bench

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 1]
```diff
@@ -394,11 +394,6 @@ theorem op_norm_comp_le (f : E →SL[σ₁₂] F) : ‖h.comp f‖ ≤ ‖h‖ *
       exact h.le_op_norm_of_le (f.le_op_norm x)⟩
 #align continuous_linear_map.op_norm_comp_le ContinuousLinearMap.op_norm_comp_le
 
--- Porting note: restatement of `op_norm_comp_le` for linear maps.
-/-- The operator norm is submultiplicative. -/
-theorem op_norm_comp_le' (h : Eₗ →L[𝕜] Fₗ) (f : E →L[𝕜] Eₗ) : ‖h.comp f‖ ≤ ‖h‖ * ‖f‖ :=
```
Comment (alexjbest) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:399]: what is the reason to delete this?

Comment (eric-wieser) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:399]: It's syntactically a special case of the above

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 2]
```diff
@@ -981,22 +971,9 @@ def prodMapL : (M₁ →L[𝕜] M₂) × (M₃ →L[𝕜] M₄) →L[𝕜] M₁
...
-      -- Porting note: mathport suggested:
-      -- ```
-      -- simp only [add_apply, coe_comp', coe_fst', Function.comp_apply, compL_apply, flip_apply,
-      --   coe_snd', inl_apply, inr_apply, Prod.mk_add_mk, add_zero, zero_add, coe_prodMap'
-      --   Prod_map, Prod.mk.inj_iff, eq_self_iff_true, and_self_iff]
-      -- rfl
-      -- ```
-      -- Frustratingly, in `mathlib3` we can use:
-      -- ```
-      -- dsimp   -- ⊢ (⇑φ x.fst, ⇑ψ x.snd) = (⇑φ x.fst + 0, 0 + ⇑ψ x.snd)
```
Comment (eric-wieser) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: Does dsimp work with the same trick? I think this proof is easier to maintain if so

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean [hunk 3]
```diff
@@ -981,22 +971,9 @@ def prodMapL : (M₁ →L[𝕜] M₂) × (M₃ →L[𝕜] M₄) →L[𝕜] M₁
...
-      -- simp
-      -- ```
-      -- Here neither `dsimp` or `simp` seem to make progress.
-      repeat first | rw [add_apply] | rw [comp_apply] | rw [flip_apply] | rw [compL_apply]
-      simp only [coe_prodMap', Prod_map, coe_fst', inl_apply, coe_snd', inr_apply, Prod.mk_add_mk,
-        add_zero, zero_add])
+      -- We have to use `simp [(X)]` to avoid unification of instance implicit arguments
+      -- which only unify at `.default` reducibility
+      simp [(add_apply), (comp_apply), (flip_apply), (compL_apply)])
```
Comment (urkud) [inline Mathlib/Analysis/NormedSpace/OperatorNorm.lean:976]: What semireducible definitions need to be unfolded?

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 2 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: Sorry I am confused as to what is being asked here. Swap out `dsimp` for `simp`?

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 3 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:976]: Lean doesn't seem to want to unfold `PseudoMetricSpace.toUniformSpace`.

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 2 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: I pushed a `simp? says`. Let me know if you think that is an improvement in this regard. (It also told me I could drop some parens).

Comment (eric-wieser) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: The lean3 proof was apparently `dsimp; simp`. I'm suggesting you restore that proof as `dsimp [(foo)]; simp [(bar)]` as appropriate, ideally reaching the same intermediate goal state.

Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: I tried that at first. Let me try again.

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:399]: I imagine this was added only to aid the port when Lean wouldn’t automatically infer the correct type. I don’t see a use for this now that things “just work.”

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 2 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: Best I can do is 
```lean
dsimp -- does some things 
simp [(add_apply), (comp_apply)]
```
in Lean 3 the `add_apply` and `comp_apply` get called with the `dsimp` leaving only `zero_add` and `add_zero` to clean up but the parens trick doesn't work here with `dsimp`. 

We could also just do 
```lean
simp [(add_apply), (comp_apply)]
```
alone. Thoughts?

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 3 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean]: Actually this might be another case of `RingHom.id` not being reducible. See[Simp claims not defeq but rfl works](https://leanprover.zulipchat.com/#narrow/stream/270676-lean4/topic/Simp.20claims.20not.20defeq.20but.20rfl.20works)

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 1 above)
Comment (alexjbest) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:399]: Yeah I was just confused as it wasn't used anywhere it seems? So why would the porter add it, oh well if it seems useless its good to get rid of it

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 3 above)
Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean]: No that doesn't help anything.

Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean]: We end up at 
```lean
 ❌ @Exists =?= PseudoMetricSpace.toUniformSpace.1.1 
```

File: Mathlib/Analysis/NormedSpace/OperatorNorm.lean (hunk 2 above)
Comment (urkud) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: I see that your PR about `Algebra.id` was merged recently. Is this workaround still needed?

Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: Unfortunately, yes.

Comment (mattrobball) [reply Mathlib/Analysis/NormedSpace/OperatorNorm.lean:993]: I found the declarations that need to be reducible.

**Comment by mattrobball** [comment]
I found the declarations that need to reducible to avoid `simp[(X)]`. This PR depends on those now.

**Comment by mattrobball** [comment]
!bench

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+
